packages/adp-tooling/src/base/prompt.ts

```diff
@@ -19,67 +23,89 @@ export type PromptDefaults = {
  * Prompt the user for the required properties for an adaptation project.
  *
  * @param defaults optional default values for the prompts
+ * @param logger optional logger instance
  * @returns a configuration for the adp writer
  */
-export async function promptGeneratorInput(defaults: PromptDefaults = {}): Promise<AdpWriterConfig> {
+export async function promptGeneratorInput(
+    defaults: PromptDefaults = {},
+    logger = new ToolsLogger()
+): Promise<AdpWriterConfig> {
+    const target = await prompts([
+        {
+            type: 'text',
+            name: 'url',
+            message: 'Target system url:',
+            initial: defaults.url,
+            validate: (input) => input?.length > 0
+        },
+        {
+            type: 'text',
+            name: 'client',
+            message: 'Client (optional):',
+            initial: defaults.client,
+            validate: (input) => (input ? input.length < 4 : true)
+        }
+    ]);
+    const provider = await createAbapServiceProvider(
+        target,
+        {
+            ignoreCertErrors: defaults.ignoreCertErrors
+        },
+        true,
+        logger
+    );
+    logger.info('Fetching system information...');
+    const ato = await provider.getAtoInfo();
+    const layer = ato.tenantType === 'SAP' ? 'VENDOR' : 'CUSTOMER_BASE';
```

Comment:
We can use here `flexLayer.VENDOR` and `flexLayer.CUSTOMER_BASE`.

Reply:
That's no cleaned up. I am not using enums but types.